controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1755,18 +1795,31 @@ func (r *BareMetalHostReconciler) createHostFirmwareComponents(info *reconcileIn
 
 			// Set bmh as owner, this makes sure the resource is deleted when bmh is deleted
 			if err = controllerutil.SetControllerReference(info.host, hfc, r.Scheme()); err != nil {
-				return errors.Wrap(err, "could not set bmh as controller")
+				return errors.Wrap(err, "could not set bmh as controller for hostFirmwareComponents")
 			}
 			if err = r.Create(info.ctx, hfc); err != nil {
 				return errors.Wrap(err, "failure creating hostFirmwareComponents resource")
 			}
 
 			info.log.Info("created new hostFirmwareComponents resource")
-		} else {
-			// Error reading the object
-			return errors.Wrap(err, "could not load hostFirmwareComponents resource")
+			return nil
 		}
+		// Error reading the object
+		return errors.Wrap(err, "could not load hostFirmwareComponents resource")
 	}
+	// Necessary in case the CRD is created manually.
+
+	if !ownerReferenceExists(info.host, hfc) {
+		if err := controllerutil.SetControllerReference(info.host, hfc, r.Scheme()); err != nil {
```

Comment:
I'd like to switch to a regular owner reference (instead of a controller reference) -both here and above on line 1797 - before we ship this. We don't want to have to add code to clean up existing controller references; better to have the flag cleared correctly from the first time it's usable.

Reply:
@zaneb you mean using controllerutil.SetOwnerReference ,  right?

Reply:
If yes, would be ok to push the change upstream separate from the follow-up discussed upstream or it should be all together?

Reply:
I think it's independent, so you could open a PR for just that upstream and add the commit here when it merges.